react/components/AutocompleteInput/index.tsx

```diff
@@ -87,7 +87,7 @@ export type AutocompleteInputProps = Omit<
   'input'
 > & {
   input: PropTypes.InferProps<typeof propTypes>['input'] &
```

Comment:
Would just changing order already solve this? So any prop we add would always overwrite the ones from input 🤔 
```suggestion
  input: React.HTMLProps<HTMLInputElement> & 
    PropTypes.InferProps<typeof propTypes>['input']
```

Reply:
Actually the problem will still be the same. It won't override. Example:
```ts
type Props = {
  selected: boolean
} & { selected: number }
```
Gives me this:
![image](https://user-images.githubusercontent.com/15948386/88330507-df1b6f80-cd01-11ea-927d-162bb84c8884.png)

Reply:
Linked version with your suggestion:

![image](https://user-images.githubusercontent.com/15948386/88330756-41747000-cd02-11ea-8ca9-ab4b55e9659a.png)

Reply:
But I'll enhance this definition to be more precise. Thanks :)